src/StripStylesTransformer.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  Node,
+  SourceFile,
+  TransformationContext,
+  Transformer,
+  Visitor,
+  Identifier,
+  ClassDeclaration,
+  PropertyAssignment
+} from 'typescript';
+import { ConfigSet } from './TransformUtils';
+
+/** Angular component decorator Styles property name */
+const STYLES = 'styles';
+/** Angular component decorator name */
+const COMPONENT = 'Component';
+/** All props to be transformed inside a decorator */
+const TRANSFORM_IN_DECORATOR_PROPS = [STYLES];
+
+/**
+ * Transformer ID
+ * @internal
+ */
+export const name = 'angular-component-strip-styles';
+
+// increment this each time the code is modified
+/**
+ * Transformer Version
+ * @internal
+ */
+export const version = 1;
+
+/**
+ * The factory of hoisting transformer factory
+ * @internal
+ */
+export function factory(cs: ConfigSet) {
+  /**
+   * Our compiler (typescript, or a module with typescript-like interface)
+   */
+  const ts = cs.compilerModule;
+
+  /**
+   * Traverses the AST down inside a decorator to a styles assignment
+   * and returns a boolean indicating if it should be transformed.
+   */
+  function isInDecoratorPropertyAssignmentToTransform(node: Node): node is ClassDeclaration {
+    return getInDecoratorPropertyAssignmentsToTransform(node).length > 0;
+  }
+
+  /**
+   * Traverses the AST down inside a decorator to a styles assignment
+   * returns it in an array.
+   */
+  function getInDecoratorPropertyAssignmentsToTransform(node: Node) {
+    if (!ts.isClassDeclaration(node) || !node.decorators) {
+      return [];
+    }
+
+    return node.decorators
+      .map(dec => dec.expression)
+      .filter(ts.isCallExpression)
+      .filter(callExpr => ts.isIdentifier(callExpr.expression) && callExpr.expression.getText() === COMPONENT)
+      .reduce((acc, nxtCallExpr) => acc.concat(
+        ...nxtCallExpr.arguments
+          .filter(ts.isObjectLiteralExpression)
+          .reduce((acc, nxtArg) => acc.concat(
+            ...nxtArg.properties
+              .filter(ts.isPropertyAssignment)
+              .filter(propAss => ts.isIdentifier(propAss.name))
+              .filter(propAss =>
+                TRANSFORM_IN_DECORATOR_PROPS.includes(
+                  (propAss.name as Identifier).text
+                )
+              )
+          ), [] as PropertyAssignment[])
+      ), [] as PropertyAssignment[])
+  }
+
+  /**
+   * Clones the styles assignment and manipulates it.
+   * @param node the property assignment to change
+   */
+  function transfromStylesAssignmentForJest(node: ClassDeclaration) {
+    const mutableNode = ts.getMutableClone(node)
+    const assignments = getInDecoratorPropertyAssignmentsToTransform(mutableNode)
+
+     assignments.forEach(assignment => {
+      switch ((assignment.name as Identifier).text) {
+        case STYLES:
+          // replace initializer array with empty array
+          assignment.initializer = ts.createArrayLiteral()
+          break
+      }
```

Comment:
I don't think break will stop forEach, the loop still continues unless that is the intention ?

Reply:
Instead a `.find` could be used to return early

Reply:
Actually there were more cases before in the `SWITCH`-statement, the `forEach` should still iterate over all the elements.

I replaced it with a simple if-statement.